src/pages/RFOX/components/Stake/StakeStatus.tsx

```diff
@@ -51,9 +55,39 @@ export const StakeStatus: React.FC<StakeRouteProps & StakeStatusProps> = ({
     [confirmedQuote.stakingAmountCryptoBaseUnit, stakingAsset?.precision],
   )
 
+  const { data: maybeSafeTx } = useSafeTxQuery({
+    maybeSafeTxHash: txId ?? undefined,
+    accountId: confirmedQuote.stakingAssetAccountId,
+  })
+
   const bodyContent: BodyContent | null = useMemo(() => {
     if (!stakingAsset) return null
 
+    if (maybeSafeTx?.isQueuedSafeTx) {
+      return {
+        key: TxStatus.Pending,
+        title: [
```

Comment:
A lot of duplicate code in these parts, too - again, not worth holding up this PR for, not blocking.